framework/src/loops/ComputeUserObjectsThread.C

```diff
@@ -245,7 +245,7 @@ ComputeUserObjectsThread::onInterface(const Elem * elem, unsigned int side, Boun
 
   std::vector<UserObject *> userobjs;
   queryBoundary(Interfaces::InterfaceUserObject, bnd_id, userobjs);
-  if (_interface_user_objects.size() == 0)
+  if (_interface_user_objects.size() == 0 && _domain_objs.size() == 0)
```

Comment:
I am curious why here we are not using `userobjs` to check but use `_interface_user_objects`? Is it possible we can query whether the domain user object requests the evaluation on this interface?